android/app/src/main/java/org/uwaterloo/subletr/pages/createlisting/CreateListingPageView.kt

```diff
@@ -606,74 +563,240 @@ fun LeaseDatePicker(state: DateRangePickerState, onClick: () -> Unit) {
+	val coroutineScope = rememberCoroutineScope()
+
+	var imageUris by remember { mutableStateOf<List<Uri?>>(ArrayList()) }
+	var hasImage by remember { mutableStateOf(false) }
+
+	var openImagePicker by rememberSaveable { mutableStateOf(false) }
+	val imagePickerBottomSheetState = rememberModalBottomSheetState(skipPartiallyExpanded = true)
+
+	val imageSelectorLauncher = rememberLauncherForActivityResult(
+		contract = ActivityResultContracts.GetMultipleContents()) {
+		imageUris = it
+		hasImage = true
+	}
+
+	val cameraLauncher = rememberLauncherForActivityResult(
+		contract = ActivityResultContracts.TakePicture(),
+	) {
+		hasImage = it
+	}
 
-			} else {
-				val source = ImageDecoder
-					.createSource(context.contentResolver,it)
-				bitmap.value = ImageDecoder.decodeBitmap(source)
+	LazyRow(
+		modifier = Modifier.fillMaxWidth(),
+		verticalAlignment = Alignment.CenterVertically,
+	) {
+		item {
+			Button(
+				modifier = Modifier
+					.size(dimensionResource(id = R.dimen.xxxxxxl))
+					.border(
+						dimensionResource(id = R.dimen.xxxs),
+						textFieldBorderColor,
+						RoundedCornerShape(dimensionResource(id = R.dimen.s))
+					),
+				shape = RoundedCornerShape(dimensionResource(id = R.dimen.s)),
+				onClick = {
+					coroutineScope.launch {
+						openImagePicker = true
+					}
+				},
+				colors = ButtonDefaults.buttonColors(
+					containerColor = textFieldBackgroundColor,
+					contentColor = Color(0xFF808080),
+				),
+			) {
+				Column(
+					modifier = Modifier,
+					horizontalAlignment = Alignment.CenterHorizontally,
+					verticalArrangement = Arrangement.spacedBy(dimensionResource(id = R.dimen.xs))
+				) {
+					Icon(
+						painter = painterResource(
+							id = R.drawable.add_filled_black_32
+						),
+						contentDescription = stringResource(id = R.string.upload_images),
+					)
+					Text(
+						text = stringResource(id = R.string.add_photo),
+						color = secondaryTextColor
+					)
+				}
 			}
 
-			bitmap.value?.let {  btm ->
-				Image(bitmap = btm.asImageBitmap(),
+			if (openImagePicker) {
+				ImagePickerBottomSheet(
+					bottomSheetState = imagePickerBottomSheetState,
+					onDismissRequest = { openImagePicker = false },
+					onTakePhotoClick = {
+						coroutineScope.launch { imagePickerBottomSheetState.hide() }.invokeOnCompletion {
+							if (!imagePickerBottomSheetState.isVisible) {
+								openImagePicker = false
+							}
+							val uri = ComposeFileProvider.getImageUri(context)
+							imageUris = listOf(uri)
+							cameraLauncher.launch(uri)
+						}
+					},
+					onChooseImageClick = {
+						coroutineScope.launch { imagePickerBottomSheetState.hide() }.invokeOnCompletion {
+							if (!imagePickerBottomSheetState.isVisible) {
+								openImagePicker = false
+							}
+							imageSelectorLauncher.launch("image/*")
+						}
+					},
+				)
+			}
+
+			if (imageUris.isNotEmpty() && hasImage) {
+				imageUris.filterNotNull().map {
+					if (Build.VERSION.SDK_INT < 28) {
```

Comment:
Did we test this? We can just set a minimum SDK version and not ...

Reply:
We probably don't need to test this since it's just a school project.
That being said, it is good practice to do this despite it being ugly since enough users are still on phones that are under < 28 SDK.
To be fair, I think our set minimum SDK might be higher than 28, but if we ever want/need to lower it, this kind of thing will be helpful.